ink-wrapper/src/extensions.rs

```diff
@@ -31,7 +31,10 @@ pub trait TypeExtensions {
 
 impl TypeExtensions for Type<PortableForm> {
     fn is_primitive(&self) -> bool {
-        matches!(self.type_def(), scale_info::TypeDef::Primitive(_))
+        matches!(
+            self.type_def(),
+            TypeDef::Primitive(_) | TypeDef::Sequence(_) | TypeDef::Tuple(_) | TypeDef::Array(_)
```

Comment:
What about arrays/tuples/sequences of non-primitive types? Like `[super::MyStruct; 3]` ?

Reply:
Good point, I thought it would just work recursively but as you point out everything inside the `event` module needs to be properly prefixed with `super::`. So I would say this fix is correct, but it's only part of the story, I think the other part will be to have `type_ref` accept an optional prefix and apply it to everything.

Reply:
Can we recursively check typedefs of these types?

Reply:
Yeah, take a look now.